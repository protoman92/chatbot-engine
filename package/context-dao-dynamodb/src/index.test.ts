import { _getContextUpdateArgs } from ".";

describe("DynamoDB context DAO", () => {
  it("Getting context update arguments should work correctly", () => {
    // Setup
    // When
    expect(
      _getContextUpdateArgs({
        a: 1,
        b: [1, 2, 3],
        c: { a: 1, b: 2, c: 3 },
        d: [{ a: 1 }, { b: 2 }, { c: 3 }],
        e: [{ a: [{ b: { c: 1 } }] }],
      })
    ).toMatchSnapshot();
  });
});
