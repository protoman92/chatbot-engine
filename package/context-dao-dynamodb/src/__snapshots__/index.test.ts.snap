// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamoDB context DAO Getting context update arguments should work correctly 1`] = `
{
  "ExpressionAttributeNames": {
    "#a": "a",
    "#b": "b",
    "#c": "c",
    "#d": "d",
    "#e": "e",
  },
  "ExpressionAttributeValues": {
    ":a": 1,
    ":b": [],
    ":b_0": 1,
    ":b_1": 2,
    ":b_2": 3,
    ":c": {},
    ":c_a": 1,
    ":c_b": 2,
    ":c_c": 3,
    ":d": [],
    ":d_0": {},
    ":d_0_a": 1,
    ":d_1": {},
    ":d_1_b": 2,
    ":d_2": {},
    ":d_2_c": 3,
    ":e": [],
    ":e_0": {},
    ":e_0_a": [],
    ":e_0_a_0": {},
    ":e_0_a_0_b": {},
    ":e_0_a_0_b_c": 1,
  },
  "UpdateExpression": "
#a = :a,
#b = if_not_exists(#b, :b),
#b[0] = :b_0,
#b = if_not_exists(#b, :b),
#b[1] = :b_1,
#b = if_not_exists(#b, :b),
#b[2] = :b_2,
#c = if_not_exists(#c, :c),
#c.#a = :c_a,
#c = if_not_exists(#c, :c),
#c.#b = :c_b,
#c = if_not_exists(#c, :c),
#c.#c = :c_c,
#d = if_not_exists(#d, :d),
#d[0] = if_not_exists(#d[0], :d_0),
#d[0].#a = :d_0_a,
#d = if_not_exists(#d, :d),
#d[1] = if_not_exists(#d[1], :d_1),
#d[1].#b = :d_1_b,
#d = if_not_exists(#d, :d),
#d[2] = if_not_exists(#d[2], :d_2),
#d[2].#c = :d_2_c,
#e = if_not_exists(#e, :e),
#e[0] = if_not_exists(#e[0], :e_0),
#e[0].#a = if_not_exists(#e[0].#a, :e_0_a),
#e[0].#a[0] = if_not_exists(#e[0].#a[0], :e_0_a_0),
#e[0].#a[0].#b = if_not_exists(#e[0].#a[0].#b, :e_0_a_0_b),
#e[0].#a[0].#b.#c = :e_0_a_0_b_c
",
}
`;
